app/api/openai/route.js

```diff
@@ -0,0 +1,32 @@
+import { OpenAIApi, Configuration } from "openai";
+import { NextResponse } from 'next/server';
+
+
+const configuration = new Configuration({
+  apiKey: process.env.OPENAI_API_KEY,
+});
+
+const openai = new OpenAIApi(configuration);
+
+export async function POST(req) {
+  console.log(req, "req");
+  const { messages } = await req.json()
```

Comment:
I am not sure about it, but `const { messages } = req.body` should work, and it is more intuitive